zaza/openstack/charm_tests/octavia/tests.py

```diff
@@ -82,7 +82,7 @@ def setUpClass(cls):
 
     def resource_cleanup(self):
         """Remove resources created during test execution."""
-        for lb in self.loadbalancers:
+        for lb in self.loadbalancers.pop():
```

Comment:
This looks odd unless I am not reading this right. Is there a nested list here?


```
In [1]: l = [1, 2]                                                              

In [2]: for i in l.pop(): 
   ...:     print(i) 
   ...:                                                                         
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-b2a6c2beab5a> in <module>
----> 1 for i in l.pop():
      2     print(i)
      3 

TypeError: 'int' object is not iterable
```

Reply:
Ha, yes that is indeed weird and not the intended behavior.

The intention is to consume the list so that subsequent calls does not attempt to do resource cleanup on the same objects multiple times.

I'll propose something different, thx for catching.